### ISSUE TITLE
Feature request: add a "switch to tab" link if the tab is already open

### ISSUE THREAD

**Issue by jasononeil** [Vincent-P/firefox-better-history#26]
If a page in my history is still open as a tab, I would like a link to switch to that tab (rather than open a new tab of the same page).

I was imagining this could either be a new link  ("switch to tab" as well as "link"), or it could replace "link":

<img width="827" alt="image" src="https://user-images.githubusercontent.com/315158/118085887-b5007e80-b3f5-11eb-8372-0c47e1d22b99.png">

One thought I had which might make this undesirable: the extension would need permission to access current tabs as well as the history. When I installed this extension I appreciated that it only required 1 permission.

_I'm happy to help!_ I haven't built a web extension before but am a web developer and happy to try contribute this if you think it is useful.

**Comment by Vincent-P** [comment]
I feel like this kind of feature is out of scope for a history extension, I don't understand why you would search for your open tabs in the history.
Is there any reason why you aren't using a tab management extension?

**Comment by jasononeil** [comment]
That is fair, my use case is a bit weird. 

I've set this extension as my homepage when I open a new tab, and am looking for a UI that focuses on history rather than "open tabs" - I tend to open many tabs and forget to close them. It's not uncommon to look for something in  my tab-bar and not be able to find it because there's 50+ tabs there and I can't read them properly (even tree-style tabs is a bit much).

The history + search in this extension is perfect for finding something that I might have visited a few days ago, or even a few hours ago, but is lost in my mess of tabs (or perhaps I closed it). Having the option to search for a page, and if it happens to still be open, switch to the tab, is a workflow I want to try.

It might be time to write my own extension :)
